lib/Analysis/Memory/MemoryLocationRange.cpp

```diff
@@ -0,0 +1,339 @@
+//===- MemoryLocationRange.cpp ---- Memory Location Range -------*- C++ -*-===//
+//
+//                       Traits Static Analyzer (SAPFOR)
+//
+// Copyright 2018 DVM System Group
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+//
+// This file provides utility analysis objects describing memory locations.
+//
+//===----------------------------------------------------------------------===//
+
+#include "tsar/Analysis/Memory/MemoryLocationRange.h"
+#include <bcl/Equation.h>
+#ifndef NDEBUG
+#include "tsar/Unparse/Utils.h"
+#include <llvm/Support/Debug.h>
+#endif
```

Comment:
Иначе не собирается.

```suggestion
#include "tsar/Analysis/Memory/MemoryLocationRange.h"
#ifndef NDEBUG
#include "tsar/Unparse/Utils.h"
#endif
#include <bcl/Equation.h>
#include <llvm/Support/Debug.h>
```